### PR TITLE
2 0 status edition

### PR DESCRIPTION
**Pull request by ngilain**
Add status edition screen
https://github.com/open-orchestra/open-orchestra/pull/1118
https://github.com/open-orchestra/open-orchestra-cms-bundle/pull/2040
https://github.com/open-orchestra/open-orchestra-model-bundle/pull/708
https://github.com/open-orchestra/open-orchestra-model-interface/pull/269

### CODE REVIEW

File: WorkflowAdminBundle/Controller/Api/StatusController.php [hunk 1]
```diff
@@ -82,21 +82,64 @@ public function deleteAction($statusId)
...
+            $request->getContent(),
+            $this->getParameter('open_orchestra_workflow_admin.facade.status_collection.class'),
+            $format
+        );
+        $statusRepository = $this->get('open_orchestra_model.repository.status');
+        $statuses = $this->get('open_orchestra_api.transformer_manager')->get('status_collection')->reverseTransform($facade);
+
+        $statusIds = array();
+        foreach ($statuses as $status) {
+            $this->denyDeleteUnlessGranted($status);
```
Comment (itkg-nanne) [inline WorkflowAdminBundle/Controller/Api/StatusController.php]: Should be harmonized with : 
https://github.com/open-orchestra/open-orchestra-cms-bundle/blob/master/UserAdminBundle/Controller/Api/UserController.php#L200

**Comment by itkg-nanne** [comment]
:+1:

File: WorkflowAdminBundle/Controller/Api/StatusController.php [hunk 2]
```diff
@@ -82,21 +82,64 @@ public function deleteAction($statusId)
...
+
+    /**
+     * @param Request $request
+     *
+     * @Config\Route("/delete-multiple", name="open_orchestra_api_status_delete_multiple")
+     * @Config\Method({"DELETE"})
+     *
+     * @return Response
+     */
+    public function deleteStatusesAction(Request $request)
```
Comment (alavieille) [inline WorkflowAdminBundle/Controller/Api/StatusController.php]: can you add a functional test for this route

File: WorkflowAdminBundle/Controller/Admin/StatusController.php [hunk 3]
```diff
@@ -48,31 +48,35 @@ public function formAction(Request $request, $statusId)
...
         $form->handleRequest($request);
-        $message = $this->get('translator')->trans('open_orchestra_backoffice.form.status.creation');
+        $message = $this->get('translator')->trans('open_orchestra_workflow_admin.form.status.creation');
 
         if ($this->handleForm($form, $message, $status)) {
             $this->dispatchEvent(StatusEvents::STATUS_CREATE, new StatusEvent($status));
-            $response = new Response('', Response::HTTP_CREATED, array('Content-type' => 'text/html; charset=utf-8'));
+            $response = new Response(
+                $message,
+                Response::HTTP_CREATED,
```
Comment (alavieille) [inline WorkflowAdminBundle/Controller/Admin/StatusController.php]: why add message in the response ?

File: WorkflowAdminBundle/Resources/public/ecmascript/OpenOrchestra/Application/View/Status/NewStatusView.js [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+
+    /**
+     * Redirect to new status view
+     *
+     * @param {mixed}  data
+     * @private
+     */
+    _redirectNewStatus(data) {
+        let message = new FlashMessage(data, 'success');
+        FlashMessageBag.addMessageFlash(message);
```
Comment (alavieille) [inline WorkflowAdminBundle/Resources/public/ecmascript/OpenOrchestra/Application/View/Status/NewStatusView.js]: i think this part isn't necessary

File: WorkflowAdminBundle/Controller/Admin/StatusController.php [hunk 5]
```diff
@@ -48,31 +48,35 @@ public function formAction(Request $request, $statusId)
...
      *
      * @return Response
      */
     public function newAction(Request $request)
     {
-        $statusClass = $this->container->getParameter('open_orchestra_model.document.status.class');
+        $statusClass = $this->getParameter('open_orchestra_model.document.status.class');
         /** @var StatusInterface $status */
         $status = new $statusClass();
         $this->denyAccessUnlessGranted(ContributionActionInterface::CREATE, $status);
```
Comment (alavieille) [inline WorkflowAdminBundle/Controller/Admin/StatusController.php]: $this->denyAccessUnlessGranted(ContributionActionInterface::CREATE, StatusInterface::ENTITY_TYPE);

**Comment by ngilain** [comment]
@itkg-nanne , @alavieille : update done

**Comment by alavieille** [comment]
:+1:

**Pull request merged**